apps/angular/signal-input/src/app/user.component.ts

```diff
@@ -19,23 +19,20 @@ const ageToCategory = (age: number): Category => {
   standalone: true,
   imports: [TitleCasePipe],
   template: `
-    {{ fullName | titlecase }} plays tennis in the {{ category }} category!!
+    {{ fullName() | titlecase }} plays tennis in the {{ category() }} category!!
   `,
   host: {
     class: 'text-xl text-green-800',
   },
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class UserComponent implements OnChanges {
-  @Input({ required: true }) name!: string;
-  @Input() lastName?: string;
-  @Input() age?: string;
+export class UserComponent {
+  name = input.required<string>();
+  lastName = input('');
+  category = input.required({
+    alias: 'age',
+    transform: (age: string): Category => ageToCategory(Number(age) ?? 0),
```

Comment:
it looks smart but I'm not a big fan of it. It's harder to read, which means harder to maintain. If someone or your future you come bacl to the component, it will take you a minute or 2 to understand what the compoennt is expecting and why you did it like that, you should prefer an age input and then a category compute. Easy to read, to maintain and to extend

Reply:
I'm sympathetic to the argument that this approach is too clever. I pushed another version.